src/reader/mod.rs

```diff
@@ -405,15 +405,15 @@ impl<R> Reader<R> {
     where
         R: XmlSource<'i, B>,
     {
-        let event = match self.parser.tag_state {
-            TagState::Init => self.read_until_open(buf, true),
-            TagState::Closed => self.read_until_open(buf, false),
-            TagState::Opened => self.read_until_close(buf),
-            TagState::Empty => self.parser.close_expanded_empty(),
-            TagState::Exit => return Ok(Event::Eof),
+        let event = match self.parser.state {
+            ParseState::Init => self.read_until_open(buf, true),
+            ParseState::Closed => self.read_until_open(buf, false),
```

Comment:
Might make sense to rename these variants as e.g. `ParseState::ClosedTag`,  `ParseState::OpenTag`